frontend/src/components/Dialogs/InputDialogAttachUrl.vue

```diff
@@ -1,73 +1,113 @@
 <template>
- <input v-if="'name' in userInput"
-    placeholder="Text to display (required)"
-    v-model="name"
-    data-test="name-input"
-  />
-  <input
-    placeholder="Paste a link (required)"
-    v-model="url"
-    data-test="link-input"
-  />
-  <textarea v-if="'comments' in userInput"
-    class="comments"
-    placeholder="Comments"
-    v-model="comments"
-    data-test="comments-text-area"
-  >
+  <input v-if="'name' in userInput" placeholder="Text to display (required)" v-model="name" data-test="name-input"
+    :class="{ 'invalid-input-highlight': highlight && !name }" />
+  <div v-if="'name' in userInput && highlight && !name" class="required-message" data-test="required">
+    *required
+  </div>
+  <input placeholder="Paste a link (required)" v-model="url" data-test="link-input"
+    :class="{ 'invalid-input-highlight': highlight && !url }" />
+  <div v-if="highlight && !url" class="required-message" data-test="required">
+    *required
+  </div>
+  <textarea v-if="'comments' in userInput" class="comments" placeholder="Comments" v-model="comments"
+    data-test="comments-text-area">
   </textarea>
 </template>
 
+
 <script>
 export default {
   name: 'input-dialog-attach-url',
-  emits: ['update:userInput'],
+  emits: ['update:userInput', 'update:allRequiredFieldsFilled', 'validationDone'],
   props: {
     userInput: {
       type: Object,
       required: true,
     },
+    runValidation: {
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      highlight: false,
+    };
   },
   computed: {
+    requiredFields() {
+      return this.userInput.requiredFields || [];
+    },
     name: {
       get() {
         return this.userInput.name;
       },
       set(value) {
-        const input = this.userInput;
-        input['name'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('name', value);
       },
     },
     url: {
       get() {
         return this.userInput.data;
       },
       set(value) {
-        const input = this.userInput;
-        input['data'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('data', value);
       },
     },
     comments: {
       get() {
         return this.userInput['comments'];
       },
       set(value) {
-        const input = this.userInput;
-        input['comments'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('comments', value);
       },
     },
+    allRequiredFieldsFilled() {
+      return this.requiredFields.every((field) => !!this[field] && this[field].length > 0);
+    },
+  },
+  watch: {
```

Comment:
Why did you decide to do the validation within the child components instead of the inputDialog.js where the state of the dialog is managed?  When the ‘confirmation(userInput)’ https://github.com/uab-cgds-worthey/rosalution/pull/90/files#diff-f5058686e21862d83e368c13b0f4ccf587074f2fd093876b28427a714b08db17R118 is called,  the validation can be done before the ‘close(userInput)’ call, and if it fails, update the state of the dialog to be invalid.  The changed state of the 'dialog' in data that has the state as props would cause the re-rerender to update to show the invalid state.

Reply:
I appreciate you trying to stick to using Watchers for this solution.  I didn't anticipate it needed to go so far as to needing to differing the callback  to be done after the next DOM update cycle. https://github.com/uab-cgds-worthey/rosalution/pull/90/files#diff-b5c0bc889b479c474f59d646d55e2afe0501fcc38086e574bc130e7b62f2176cR76. 

After taking time to look at it in more detail; let's consider this exploratory work and document it.  This code would be good to be contained in Rosalution's "/docs/prototypes/" directory with its own subproject.   I will follow up with a change request to document it.